examples/http/aws_cloudwatch_metric.js

```diff
@@ -0,0 +1,292 @@
+ * https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html
+ * 
+ * The driver uses the script item to make HTTP requests to the CloudWatch API.
+ * Communication protocol is https
+ */
+
+var accessKey = D.device.username(); //accessKey == username
+var secretKey = D.device.password(); //secretKey == password
+var region = "us-east-2"; // change this by you aws region
+var requestPeriod = 600;
+var metrics;
+
+var table = D.createTable("ClouWatch", [
+    { label: "Namespace" },
+    { label: "Dimension" },
+    { label: "Dimension Value" },
+    { label: "Metric" },
+    { label: "Period" },
+    { label: "Stat" },
+    { label: "Value" },
+    { label: "Unit" }
+]);
+
+//These functions are used to compute hash-based message authentication codes (HMAC) using a specified algorithm.
+function sha256(message) {
+    return D.crypto.hash(message, "sha256", null, "hex");
+}
+function hmac(algo, key, message) {
+    key = D._unsafe.buffer.from(key);
+    return D.crypto.hmac(message, key, algo, "hex");
+}
+
+function sign(key, message) {
+    var hex = hmac("sha256", key, message);
+    if ((hex.length % 2) === 1) {
+        throw "Invalid length of a hex string!";
+    }
+    var result = new Int8Array(hex.length / 2);
+    for (var i = 0, b = 0; i < hex.length; i += 2, b++) {
+        result[b] = parseInt(hex.substring(i, i + 2), 16);
+    }
+    return result;
+}
+
+
+// this function convert json object to aws application/x-www-form-urlencoded content-type
+function prepareObject(prefix, param) {
+    var result = {};
+    if (typeof param === "object") {
+        if (Array.isArray(param)) {
+            param.forEach(function (value, index) {
+                var nested = prepareObject(prefix + ".member." + (index + 1), value);
+                Object.keys(nested).forEach(function (key) {
+                    result[key] = nested[key];
+                });
+            });
+        } else {
+            Object.keys(param).forEach(function (k) {
+                var nested = prepareObject(prefix + "." + k, param[k]);
+                Object.keys(nested).forEach(function (key) {
+                    result[key] = nested[key];
+                });
+            });
+        }
+    } else {
+        result[prefix] = param;
+    }
+    return result;
+}
+
+/**
+ * @returns CloudWatch metrics to be monitored for an AWS CloudWatch.
+ */
+function prepareParams(params) {
+    var result = [];
+    Object.keys(params).sort().forEach(function (key) {
+        if (typeof params[key] !== "object") {
+            result.push(key + "=" + encodeURIComponent(params[key]));
+        }
+        else {
+            result.push(prepareObject(key, params[key]));
+        }
+    });
+    return result.join("&");
+}
+
+// This is the metrics list, should be changed to the user need. (follow the structure in the following variable)
+var cloudWatchMetric = [
```

Comment:
Since this is the parametarizable part Maybe best to be on the top of the driver